mlir/Makefile

```diff
@@ -107,9 +113,10 @@ enzyme:
 		-DCMAKE_C_COMPILER_LAUNCHER=$(COMPILER_LAUNCHER) \
 		-DCMAKE_CXX_COMPILER_LAUNCHER=$(COMPILER_LAUNCHER) \
 		-DCMAKE_EXE_LINKER_FLAGS=$(USE_SANITIZER_FLAGS) \
-		-DCMAKE_CXX_VISIBILITY_PRESET=$(SYMBOL_VISIBILITY)
+		-DCMAKE_CXX_VISIBILITY_PRESET=$(SYMBOL_VISIBILITY) \
+		-DCMAKE_POLICY_DEFAULT_CMP0116=NEW
```

Comment:
How did this come about? :)

Reply:
Is it to get rid of this warning when building enzyme.
```python
CMake Warning (dev) at /home/tzung-han.juang/Workspace/catalyst-update/mlir/llvm-project/build/lib/cmake/llvm/TableGen.cmake:134 (add_custom_command):
  Policy CMP0116 is not set: Ninja generators transform DEPFILEs from
  add_custom_command().  Run "cmake --help-policy CMP0116" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.
Call Stack (most recent call first):
  Enzyme/CMakeLists.txt:17 (tablegen)
  Enzyme/CMakeLists.txt:24 (enzyme_tablegen)
This warning is for project developers.  Use -Wno-dev to suppress it.

```